packages/http-client/src/index.ts

```diff
@@ -7,23 +7,20 @@ export const loadProject = async ({
 }: {
   apiUrl: string;
   projectId: string;
-}): Promise<Array<CanvasData> | string> => {
-  try {
-    if (apiUrl === undefined) {
-      throw new Error("Webstudio API URL is required.");
-    }
-    const baseUrl = new URL(apiUrl);
-    const projectUrl = new URL(`/rest/project/${projectId}`, baseUrl);
-    const projectResponse = await fetch(projectUrl);
-    const project = await projectResponse.json();
-    if (!projectResponse.ok) {
-      throw new Error(project);
-    }
-    return project;
-  } catch (error) {
-    if (error instanceof Error) {
-      return error.message;
-    }
-    throw error;
+}): Promise<Array<CanvasData>> => {
+  if (apiUrl === undefined) {
+    throw new Error("Webstudio API URL is required.");
   }
+  const baseUrl = new URL(apiUrl);
+  const projectUrl = new URL(`/rest/project/${projectId}`, baseUrl);
+  const projectResponse = await fetch(projectUrl);
+
+  if (projectResponse.ok) {
+    return await projectResponse.json();
+  }
+
+  // In the case where the status code is not 2xx,
+  // we don't know what to do with the response other than just show an error message.
+  const message = await projectResponse.text();
+  throw new Error(message.substring(0, 1000));
```

Comment:
Are you trying to limit the message because we are posting entire stack trrace?

Reply:
No, often it's HTML content of intermediate proxies, like Cloudflare 522 page or like. The first 1k symbols are enough to get what's happening. Otherwise you usually ends up with huuuge log entries.